compiler/il/OMRBlock.cpp

```diff
@@ -1207,6 +1213,23 @@ static TR::SymbolReference * createSymRefForNode(TR::Compilation *comp, TR::Reso
  * Splits the blocks from the split point manually handling the GlRegDeps created by the Global Register Allocator.
  * If there are available global registers which can be used, it will utililze them to uncommon nodes across split points.
  * 
+ * @details
+ * It uses following simplified algorithm to split blocks and uncommon nodes. 
+ * 1. Analyze trees before the split point to gather information about nodes which are referenced in new block and needs replacement after split.
+ * 2. Now as it has information about all the nodes which are required to be uncommoned, it starts cretaing replacment nodes as per following rules.
+ *    a. If node to be uncommoned is regLoad / load of constant then replacement node will be copy of the original node and also mark the used register unavailable. 
+ *    b. If node to be uncommoned is stored into a register after split point and that register is available, then replacement node will be regLoad from that
+ *       register. It also makes sure that regStore tree is moved to before split point also marks used register unavailable.
+ *    c. If node to be uncommoned is stored into a register before split point and node does not succeeds in [b] case and register is available then replacement
+ *       node will be regLoad from that register also marks used register unavailable.
+ * 3. While checking for case-b and case-c from Step 2 it also checks regStores and PassThrough nodes after split point. 
+ *    a. If node under pass through requires to be uncommoned and uses different register then the replacement node
+ *       then it stores replacement node into a register used by passThrough before the tree containing that passthrough node.
+ *    b. If node under regStore does not need to be uncommoned, then mark the used register unavailable. 
+ * 4. For the remaining of nodes to be uncommoned, it tries to find the available register using list of unavailable registers and if it can find one, then stores node
```

Comment:
How many "available" registers can end up being used ? i.e. not every register available on a platform is offered up for GRA and so I am asking if the max number of available registers that can end up being used is a) the max number of registers offered for GRA or b) the max number of registers available on the platform ?

Reply:
It is `b` . Max number of registers offered by GRA